worldedit-core/src/main/java/com/sk89q/worldedit/extent/clipboard/io/ClipboardFormats.java

```diff
@@ -0,0 +1,69 @@
+package com.sk89q.worldedit.extent.clipboard.io;
+
+import java.io.File;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import javax.annotation.Nullable;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+public final class ClipboardFormats {
+
+    private static final Map<String, ClipboardFormat> aliasMap = new HashMap<>();
+    private static final List<ClipboardFormat> registeredFormats = new ArrayList<>();
+
+    public static void registerClipboardFormat(ClipboardFormat format) {
+        checkNotNull(format);
+
+        for (String key : format.getAliases()) {
+            aliasMap.put(key.toLowerCase(Locale.ENGLISH), format);
```

Comment:
I'd be a fan of an exception on conflict in this case.

---- On Tue, 16 Aug 2016 17:02:16 -0700 notifications@github.com wrote ----

In worldedit-core/src/main/java/com/sk89q/worldedit/extent/clipboard/io/ClipboardFormats.java:

> +import java.util.Locale;
> +import java.util.Map;
> +import javax.annotation.Nullable;
> +
> +import static com.google.common.base.Preconditions.checkNotNull;
> +
> +public final class ClipboardFormats {
> +
> +    private static final Map<String, ClipboardFormat> aliasMap = new HashMap<>();
> +    private static final List<ClipboardFormat> registeredFormats = new ArrayList<>();
> +
> +    public static void registerClipboardFormat(ClipboardFormat format) {
> +        checkNotNull(format);
> +
> +        for (String key : format.getAliases()) {
> +            aliasMap.put(key.toLowerCase(Locale.ENGLISH), format);
Not inaccessible - see getAll. But perhaps we should throw on conflict?

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub, or mute the thread.